Assets/Scripts/Entities/PrincessFrog.cs

```diff
@@ -0,0 +1,110 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace Entities
+{
+    public class PrincessFrog : EnemyClass
+    {
+        /* This enemy triggers changes in the scene layout by spawning enemies. */
+        private SceneBuilder.SceneBuilder _sceneBuilder;
```

Comment:
Lets decouple the scene builder from the princess frog. Maybe entities can have a "OnEntitySpawned" event that fires after all of its initialization code in Start() is complete.